plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java

```diff
@@ -257,4 +261,41 @@ public PinotConfig setMaxRowsPerSplitForSegmentQueries(int maxRowsPerSplitForSeg
         this.maxRowsPerSplitForSegmentQueries = maxRowsPerSplitForSegmentQueries;
         return this;
     }
+
+    public int getMaxRowsForBrokerQueries()
+    {
+        return maxRowsForBrokerQueries;
+    }
+
+    @Config("pinot.max-rows-for-broker-queries")
+    public PinotConfig setMaxRowsForBrokerQueries(int maxRowsForBrokerQueries)
+    {
+        this.maxRowsForBrokerQueries = maxRowsForBrokerQueries;
+        return this;
+    }
+
+    public boolean isAggregationPushdownEnabled()
+    {
+        return aggregationPushdownEnabled;
+    }
+
+    @Config("pinot.aggregation-pushdown.enabled")
+    public PinotConfig setAggregationPushdownEnabled(boolean aggregationPushdownEnabled)
+    {
+        this.aggregationPushdownEnabled = aggregationPushdownEnabled;
+        return this;
+    }
+
+    public boolean isDistinctCountPushdownEnabled()
+    {
+        return distinctCountPushdownEnabled;
+    }
+
+    @Config("pinot.distinct-count-pushdown.enabled")
+    @ConfigDescription("Controls whether distinct count is pushed down to Pinot. Distinct count pushdown can cause Pinot to do a full scan. Aggregation pushdown must also be enabled in addition to this parameter otherwise no pushdowns will be enabled.")
```

Comment:
Add validation to disallow setting this without also setting `pinot.aggregation-pushdown.enabled`.

Btw why do we need this? If we don't pushdown `COUNT(DISTINCT)` then also we might end up with a full-scan? Or is that not true? i.e. why can't we have just `pinot.aggregation-pushdown.enabled`?

Reply:
If count-distinct-pushdown is enabled it may crash pinot servers due to full scan. If it is disabled and there are more trino workers then it could actually return faster since the workers will do the count and the results will stream from pinot.

Reply:
Can you add a `@PostConstruct public void validate()` to validate that this is not set without also setting `pinot.aggregation-pushdown.enabled`?

Reply:
Oh nice, thanks!